### PR TITLE
Export custom `RefreshControl` component and add logic required to handle it

### PR DESCRIPTION
**Pull request by j-piasecki**
## Description

Closes https://github.com/software-mansion/react-native-gesture-handler/issues/1067.

The problem with `RefreshControl` on Android was that it wasn't a part of the Gesture Handler system. This PR adds a custom `RefreshControl` component exported by GH in the same way as `ScrollView`, `Switch` etc. It also adds custom logic to `ScrollView` and `FlatList` components exported by Gesture Handler that handles setting relations between native gestures properly.

## Test plan

Tested on the Example app and on code [from the issue](https://github.com/software-mansion/react-native-gesture-handler/issues/1067#issuecomment-1013765946)

### CODE REVIEW

File: src/components/GestureComponents.tsx [hunk 1]
```diff
@@ -56,9 +94,13 @@ export type DrawerLayoutAndroid = typeof DrawerLayoutAndroid &
   RNDrawerLayoutAndroid;
 
 export const FlatList = React.forwardRef((props, ref) => {
+  const refreshControlGestureRef = React.useRef<any>();
```
Comment (tomekzaw) [inline src/components/GestureComponents.tsx:97]: Why `any`?

File: src/components/GestureComponents.tsx [hunk 2]
```diff
@@ -70,14 +112,35 @@ export const FlatList = React.forwardRef((props, ref) => {
...
       ref={ref}
       {...flatListProps}
       renderScrollComponent={(scrollProps) => (
-        <ScrollView {...{ ...scrollProps, ...scrollViewProps }} />
+        <ScrollView
+          {...{
+            ...scrollProps,
+            ...scrollViewProps,
+            waitFor: [
+              ...toArray(waitFor ? waitFor : []),
```
Comment (tomekzaw) [inline src/components/GestureComponents.tsx:127]: What about `waitFor ?? []`?

File: src/components/GestureComponents.tsx [hunk 3]
```diff
@@ -25,16 +26,53 @@ import {
...
+>((props, ref) => {
+  const refreshControlGestureRef = React.useRef<any>();
+  const { refreshControl, waitFor, ...rest } = props;
+
+  return (
+    <GHScrollView
+      {...rest}
+      // @ts-ignore `ref` exists on `GHScrollView`
+      ref={ref}
+      waitFor={[...toArray(waitFor ? waitFor : []), refreshControlGestureRef]}
```
Comment (tomekzaw) [inline src/components/GestureComponents.tsx:57]: What about `waitFor ?? []`?

Comment (j-piasecki) [reply src/components/GestureComponents.tsx]: Done in https://github.com/software-mansion/react-native-gesture-handler/pull/2137/commits/5571228feea976af1751e93422f8e8d82f30b548

File: src/components/GestureComponents.tsx (hunk 2 above)
Comment (j-piasecki) [reply src/components/GestureComponents.tsx]: Done in https://github.com/software-mansion/react-native-gesture-handler/pull/2137/commits/5571228feea976af1751e93422f8e8d82f30b548

File: src/components/GestureComponents.tsx (hunk 1 above)
Comment (j-piasecki) [reply src/components/GestureComponents.tsx]: Changed the type and the value in https://github.com/software-mansion/react-native-gesture-handler/pull/2137/commits/5571228feea976af1751e93422f8e8d82f30b548

**Review by tomekzaw — APPROVED** [review]

**Pull request merged**